src/sipmfit.jl

```diff
@@ -104,9 +104,11 @@ function fit_sipm_spectrum(pe_cal::Vector{<:Real}, min_pe::Real=0.5, max_pe::Rea
 
     # get pe_pos
     get_pe_pos = pe -> dot(μ[in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)], w[in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)] ) / sum(w[in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)])
+    get_pe_res = pe -> dot(σ[in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)], w[in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)] ) / sum(w[in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)])
```

Comment:
It looks like you are calculating this 3 times in each call:
```julia
in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)
```

Could you do something like (not tested though):
```julia
get_pe_res = pe -> begin
    sel = in.(μ, (-Δpe_peak_assignment..Δpe_peak_assignment) .+ pe)]
    dot(view(σ,sel), view(w,sel)) / sum(view(w,sel))
end 
```
If you work with `view`, you might be able to avoid some allocations.

Reply:
Fixed